grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -88,6 +88,13 @@ class LocationController {
             if (locationInstance?.id && supportedActivities?.empty) {
                 params.supportedActivities = [ActivityCode.NONE.id]
             }
+
+            // Map supportedActivities to mark it as dirty for save,
+            // supportedActivities cannot be empty to avoid overriding ActivityCode.NONE
+            if (locationInstance?.id && !supportedActivities?.empty) {
```

Comment:
as for `update` I might be wrong, because we have `save` (for creating new) and `update` for editing, but for some we have `update` for both - create and edit.
Just check when creating a location if it uses the `update`. If it does, then checking the id could've had a sense.

Reply:
if checking the `id` will turn out to be expected, then redo my preposition of shoretning to:
```groovy
if (locationInstance.id) {
 params.supportedActivities = supportedActivities?.empty ? [ActivityCode.NONE.id] : supportedActivities
}

Reply:
I checked if this method is used for creating location also - yes, it is